### DISCUSSION TITLE
Module to create C4 Model Visualizations

### DISCUSSION THREAD

**Discussion by mrk007dvt** [IntentArchitect/Support#20] [Ideas]
Post a project creation, is there a module that can create a basic C4 Model Visualization using any of the supporting tooling such as C4-PlantUML, Structurizr, and so on. Even if a complete visualization is not possible, just adding the components would be useful.

[https://c4model.com/#Tooling](url)

**Comment by joelsteventurner** [comment]
Hi @mrk007dvt 

This is a very interesting idea and a great suggestion. We don’t currently have a module which does this, but one could be created. 

Some of these diagrams would be straight forward to produce, some would require some additional metadata to be modelled.

Did you have any specific requirements i.e. any diagrams you are specifically interested in or is it a more that you like to have c4 suite of diagrams for documentation purposes.

I have created a ticket for this on our backlog to be discussed and action and I will revert to you.

**Reply by mrk007dvt** [reply]
Thanks. There is no specific requirement. It seems redundant to design in Visio and then again in IA. It would be good to do the design once in IA, and get a printout of the documentation, and be able to automatically update the documentation as code changes.

**Comment by joelsteventurner** [comment]
@mrk007dvt I would agree, one of the great features of Intent Architect is the Visual Design which is in sync with the code base.
